src/util/http/websocket/QueryId.h

```diff
@@ -111,5 +124,31 @@ class QueryRegistry {
     } while (!result.has_value());
     return std::move(result.value());
   }
+
+  /// Member function that acquires a read lock and returns a vector
+  /// of all currently registered queries.
+  std::vector<QueryId> getActiveQueries() const {
+    return registry_->withReadLock([](const auto& map) {
+      // TODO<C++23> Use ranges to transform map keys into vector
+      std::vector<QueryId> result;
+      result.reserve(map.size());
+      for (const auto& entry : map) {
+        result.emplace_back(entry.first);
+      }
+      return result;
+    });
+  }
+
+  /// Returns the cancellation handle from the registry if it exists, nullptr
+  /// otherwise.
+  SharedCancellationHandle getCancellationHandle(const QueryId& queryId) const {
+    return registry_->withReadLock(
+        [&queryId](const auto& map) -> SharedCancellationHandle {
+          if (map.contains(queryId)) {
+            return map.at(queryId);
+          }
+          return nullptr;
+        });
```

Comment:
```suggestion
       auto it = map.find(queryId);
       return it != map.end() ? it->second : nullptr
```
(only perform the lookup once). Should be a default pattern to get used to, even if performance in this concrete case is probably not that crucial).